src/runtime_src/core/edge/drm/zocl/zocl_kds.c

```diff
@@ -115,30 +252,26 @@ zocl_add_context(struct drm_zocl_dev *zdev, struct kds_client *client,
 	}
 
 	mutex_lock(&client->lock);
-	if (!client->num_ctx) {
-		ret = zocl_lock_bitstream(zdev, id);
-		if (ret)
+
+	cctx = zocl_check_exists_context(client, id);
+	if (cctx == NULL) {
+		/* No existing context found. Create a new context to this client */
+		cctx = zocl_create_client_context(zdev, client, id);
+		if (cctx == NULL)
 			goto out;
```

Comment:
Probably need to return an error if client context creation failed.

Reply:
Will add this for next PR. Thanks